recipes/litehtml/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+import os
+import textwrap
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class LitehtmlConan(ConanFile):
+    name = "litehtml"
+    description = "litehtml is the lightweight HTML rendering engine with CSS2/CSS3 support."
+    license = "BSD-3-Clause"
+    topics = ("render engine", "html", "parser")
+    homepage = "https://github.com/litehtml/litehtml"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _min_cppstd(self):
+        return "11"
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._min_cppstd)
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("litehtml must be built as static on windows")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
```

Comment:
https://github.com/litehtml/litehtml/blob/6236113734bb0a28467e5999e86fdd2834be8e01/CMakeLists.txt#L154
Add an option for this?
```suggestion
        self._cmake.definitions["LITEHTML_UTF8"] = False
        self._cmake.definitions["BUILD_TESTING"] = False
```

Reply:
I'm not sure what the policy on this is? Shouldn't we leave this as is, if upstream decides to change the default shouldn't a new cci version follow that new default?

Reply:
We usually make the conan option follow the upstream default.
But a FIXME/TODO may suffice here as well.

Reply:
Added

Reply:
Thanks! Can you mark these conversations as resolved?